pkg/binding/placement.go

```diff
@@ -124,6 +125,22 @@ func (c *Controller) getPlacementByName(name string) (runtime.Object, error) {
 	return got, nil
 }
 
+func (c *Controller) evaluateBindingPolicies(ctx context.Context) {
+	c.logger.Info("evaluating BindingPolicies")
+	placements, err := c.listPlacements()
+	if err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
+	for _, placement := range placements {
+		err = c.handlePlacement(ctx, placement)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return
+		}
+	}
+}
```

Comment:
the right thing to do is to push the placement key to the workqueue.
otherwise you might have multiple threads of this controller working on the same placement in parallel.
in such a scenario in classic controllers - the key of placement is pushed to the workqueue and later is popped by a single worker thread and being updated. if we keep this principal it guarantees only a single worker works on a placement at any given moment.

Reply:
Although this kills the optimization in my comment above, it probably is the right thing to do.

On the other hand, the resolver being thread safe and the evaluation of the where being "correct at the time of call" would make this work, if I'm not missing something. 

Can go with the safe path and optimize later.

Reply:
Done